website/addons/base/views.py

```diff
@@ -435,6 +435,7 @@ def addon_view_file(auth, node, node_addon, guid_file, extras):
         # Note: must be called after get_or_start_render. This is really only for github
         'size': size,
         'extra': json.dumps(getattr(guid_file, 'extra', {})),
+        'source_url': getattr(guid_file, 'source_url', None),
```

Comment:
After seeing that you've defined it for several subclasses of `GuidFile`, please go ahead and make it a property of the parent class and remove this logic.